pkg/loki/loki.go

```diff
@@ -822,9 +822,9 @@ func (t *Loki) setupModuleManager() error {
 		QueryFrontendTripperware:     {Server, Overrides, TenantConfigs},
 		QueryFrontend:                {QueryFrontendTripperware, Analytics, CacheGenerationLoader, QuerySchedulerRing, UIRing},
 		QueryScheduler:               {Server, Overrides, MemberlistKV, Analytics, QuerySchedulerRing, UIRing},
-		QueryEngine:                  {QueryEngineScheduler},
-		QueryEngineScheduler:         {Server, Overrides, TenantConfigs, Analytics},
+		QueryEngine:                  {QueryEngineWorker},
 		QueryEngineWorker:            {QueryEngineScheduler},
```

Comment:
these dependencies do not look right to me:
`QueryEngineWorker` depending on `QueryEngineScheduler`
`QueryEngine` depending on `QueryEngineWorker`

I don't know if we need the `QueryEngine` module, maybe useful in query-frontend to create the workflow

Reply:
I was pretty confused about that as well, but when running a worker with local transport then the scheduler needs to be injected into the worker, therefore this dependency.

Eventually, we can get rid of the QueryEngine module, but it is still used in the Querier at the moment.